### PR TITLE
(fix): move zarr import inside try-catch in test helpers

### LINKED ISSUE #1342: We broke scanpy’s tests
### Please make sure these conditions are met

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the latest version of anndata.
- [X] (optional) I have confirmed this bug exists on the master branch of anndata.

### Report

The tests were broken in https://github.com/scverse/anndata/pull/1266

```console
pip install anndata  # anndata[testing] doesn‘t exist
```

```python
import anndata.tests.helpers
```

Traceback:

```pytb
ModuleNotFoundError: No module named 'zarr'
```

It’s time to finally fix #646

### Versions

```python
anndata 0.10.5
```

### PR DESCRIPTION
**Pull request by ilan-gold**
Is there some sort of way to test this?  Open to suggestions.  `ruff` does not support custom rules, I think?

- [x] Towards #1342
- [ ] Tests added
- [x] Release note added (or unnecessary)

### CODE REVIEW

File: anndata/tests/helpers.py [hunk 1]
```diff
@@ -759,21 +758,30 @@ def shares_memory_sparse(x, y):
...
+        def set_key_trackers(self, keys_to_track):
+            for k in keys_to_track:
+                self._access_count[k] = 0
+except ImportError:
+
+    class AccessTrackingStore:
+        def __init__(self) -> None:
+            raise ImportError(
+                "zarr must be imported to create an `AccessTrackingStore` instance."
+            )
```
Comment (ilan-gold) [inline anndata/tests/helpers.py:787]: Not a huge fan of this, but I think the other option is to create `AccessTrackingStore` based on some dummy base class, which will error out on construction with a less helpful error.  And if we want to raise an error, we're sort of back to this.

**Comment by ivirshup** [comment]
I think an `anndata-mindeps` test job would be the way to test this, and then testing all the functionality used by `scanpy-mindeps` uses under that job. I don't think a static check would be enough here, since a something could always be imported dynamically

File: docs/release-notes/0.10.6.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ```{rubric} Bugfix
 ```
+* Do not raise `ImportError` when `zarr` is imported in `anndata.tests.helpers` due to `scanpy` relying on functionality there. {pr}`1343` {user}`ilan-gold`
```
Comment (ivirshup) [inline docs/release-notes/0.10.6.md:5]:
```suggestion

* Defer import of zarr in test helpers, as scanpy CI job relies on them {pr}`1343` {user}`ilan-gold`
```

I'd prefer not advertising the private module.

File: anndata/tests/helpers.py [hunk 3]
```diff
@@ -759,21 +758,30 @@ def shares_memory_sparse(x, y):
...
-    def set_key_trackers(self, keys_to_track):
-        for k in keys_to_track:
-            self._access_count[k] = 0
+        def set_key_trackers(self, keys_to_track):
+            for k in keys_to_track:
+                self._access_count[k] = 0
+except ImportError:
+
+    class AccessTrackingStore:
+        def __init__(self) -> None:
```
Comment (flying-sheep) [inline anndata/tests/helpers.py:784]:
```suggestion
        def __init__(self, *_args, **_kwargs) -> None:
```

**Review by flying-sheep — CHANGES_REQUESTED** [review]

**Review by ivirshup — APPROVED** [review]

**Pull request merged**